source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.h

```diff
@@ -60,9 +60,13 @@ class GrpcClientImpl : public Client,
                  Tracing::Span& span) override;
 
 private:
-  void toAuthzResponseHeader(
-      ResponsePtr& response,
-      const Protobuf::RepeatedPtrField<envoy::config::core::v3::HeaderValueOption>& headers);
+  using RepeatedHeaderValueOption =
+      Protobuf::RepeatedPtrField<envoy::config::core::v3::HeaderValueOption>;
+  void copyHeaderMutationsIntoResponse(
```

Comment:
One thing to confirm: The validation (which is introduced by this function) is only applied to gRPC client, right?

Reply:
No, the http client validates headers to remove.

Reply:
1. In the http client, headers to add / append are expressed via header names over raw http. If the side stream were to express an invalid header in this way, they would just create an invalid http response.
2. I added validation for headers_to_remove because it gets turned into a LowerCaseString. If LowerCaseString::valid fails, it causes an assertion failure. Thus we cannot fuzz the side stream response.

I'm fine with not adding validation to the http client in this PR. The main thing I want to get out of this is fuzzing the grpc response.

Reply:
http client changes have been removed from this PR.